lib/grape/dsl/inside_route.rb

```diff
@@ -29,7 +29,10 @@ module PostBeforeFilter
         def declared(params, options = {}, declared_params = nil)
           options = options.reverse_merge(include_missing: true, include_parent_namespaces: true)
 
-          declared_params ||= (!options[:include_parent_namespaces] ? route_setting(:declared_params) : (route_setting(:saved_declared_params) || [])).flatten(1) || []
+          all_declared_params = route_setting(:declared_params)
+          current_namespace_declared_params = route_setting(:saved_declared_params).last
+
+          declared_params ||= options[:include_parent_namespaces] ? all_declared_params : current_namespace_declared_params
```

Comment:
I think it won't harm to provide a short explanation what is going on here.

`route_setting(:declared_params)` keep every parameter in a flat array, for example:
```
route_setting(:declared_params) # [:parent_name, :child_name, :child_age, :grandchild_name]
```
However, `route_setting(:saved_declared_params)` stores parameters in groups by namespace.
```
route_setting(:saved_declared_params) # [[:parent_name], [:child_name, :child_age], [:grandchild_name]]
```
The last element here contains only parameters that were defined in current namespace.